bin/sidecli.re

```diff
@@ -392,6 +427,137 @@ let inject_genesis = {
   );
 };
 
+// Create files needed for the node's operation
+let info_setup_node = {
+  let doc = "Creates the files needed to setup a node.";
+  Term.info("setup-node", ~version="%‌%VERSION%%", ~doc, ~exits, ~man);
+};
+
+let setup_node =
+    (
+      folder,
+      secret',
+      uri,
+      tezos_rpc_node,
+      _tezos_secret,
+      tezos_consensus_contract,
+    ) => {
+  let in_folder = Filename.concat(folder);
+
+  let secret = {
+    switch (secret') {
+    | Tezos_interop.Secret.Ed25519(secret) => secret
+    };
+  };
+  let address = Address.of_key(secret);
+  let (identity, identity_path) = (
+    State.{key: secret, uri, t: address},
+    "identity.json" |> in_folder,
+  );
+
+  let (wallet, wallet_path) = (
+    Files.Wallet.{address: Wallet.of_address(address), priv_key: secret},
+    "wallet.json" |> in_folder,
+  );
+
+  let (validators, validators_path) = ([], "validators.json" |> in_folder); // TODO: create validators file. I think this requires peer discovery?
+  let (interop_context, interop_context_path) = (
+    Tezos_interop.Context.{
+      rpc_node: tezos_rpc_node,
+      secret: secret',
+      consensus_contract: tezos_consensus_contract,
+      required_confirmations: 30 // TODO: What value to use here? I think Tezos finality is 30 blocks
```

Comment:
https://github.com/marigold-dev/sidechain/issues/73#issue-954177362

I mentioned here, `required_confirmations` should be 10 out of the box